src/main/java/zd/zero/waifu/motivator/plugin/assets/RemoteAssetDefinitionService.kt

```diff
@@ -9,24 +11,33 @@ abstract class RemoteAssetDefinitionService<T : AssetDefinition, U : Asset>(
 ) {
     private val random = Random(System.currentTimeMillis())
 
-    fun getRandomAssetByCategory(waifuAssetCategory: WaifuAssetCategory): Optional<U> =
-        pickRandomAsset(remoteAssetManager.supplyAssetDefinitions(), waifuAssetCategory)
-            .flatMap { randomRemoteAsset ->
-                remoteAssetManager.resolveAsset(randomRemoteAsset)
-            }
-            .map { it.toOptional() } // todo: replace with or when supporting only JRE 11+
-            .orElseGet {
-                pickRandomAsset(remoteAssetManager.supplyLocalAssetDefinitions(), waifuAssetCategory)
-                    .flatMap { randomLocalAsset ->
-                        remoteAssetManager.resolveAsset(randomLocalAsset)
-                    }
+    fun getRandomAssetByCategory(
+        waifuAssetCategory: WaifuAssetCategory
+    ): Optional<U> =
+        pickRandomAsset(remoteAssetManager.supplyLocalAssetDefinitions(), waifuAssetCategory)
+            .map {
+                downloadNewAsset(waifuAssetCategory)
+                remoteAssetManager.resolveAsset(it)
+            }.orElseGet {
```

Comment:
Get the current local assets by category and download a new remaining asset from remote by category.